homeassistant/components/tessie/lock.py

```diff
@@ -48,3 +53,32 @@ async def async_unlock(self, **kwargs: Any) -> None:
         """Set new value."""
         await self.run(unlock)
         self.set((self.key, False))
+
+
+class TessieCableLockEntity(TessieEntity, LockEntity):
+    """Cable Lock entity for Tessie."""
+
+    def __init__(
+        self,
+        coordinator: TessieStateUpdateCoordinator,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator, "charge_state_charge_port_latch")
+
+    @property
+    def is_locked(self) -> bool | None:
+        """Return the state of the Lock."""
+        return self._value == TessieChargeCableLockStates.ENGAGED
+
+    async def async_lock(self, **kwargs: Any) -> None:
+        """Charge cable Lock cannot be manually locked."""
+        raise HomeAssistantError(
+            "Insert cable to lock",
+            translation_domain=DOMAIN,
+            translation_key="no_cable",
+        )
```

Comment:
I think we should use `ServiceValidationError` instead here. This would now cause a stack trace in the log file. What do you think?

Reply:
That sounds much better. I will adjust to that.